### DISCUSSION TITLE
Help finding free resource like open street map like images that can be saved locally to use offline

### DISCUSSION THREAD

**Discussion by nixongr14** [pelicanmapping/osgearth#2437] [Support Q&A]
I have no problem using open street maps XYZ in my earth file, however that only works if I have access to online.

What I'd like to do is save a section of an open street map locally that I can then import offline. 

Anyone know where/how I can do that? Doesn't need to be open street maps specfically. I just would like to be able to overlay my map with roads and city names if possible.

Here's my example of doing this when I had access to online server :
 

![Capture](https://github.com/gwaldron/osgearth/assets/50470832/cd7f4dcb-ec92-4340-add6-c40ccfb943c7)

**Comment by gwaldron** [comment]
You could use the `osgearth_conv` tool to download the tiles you need to a local file.

For example, given the example earth file `osm.earth`, you can say
```
osgearth_conv --in-earth osm.earth --in-layer "OSM Imagery" --out driver mbtilesimage --out format jpg --out url osm.mbtiles --max-level 7
```
Of course, the whole dataset would take an enormous amount of time and disk space, so use the command line arguments to only convert what you need.

**Reply by nixongr14** [reply]
Thanks again Glenn!

I do have a follow-up. I've been trying to piece together information on my own, but how exactly do I use the osgearth_conv tool ? Honestly just need a README or simple step 1 download this / step 2 do this .

The tool seems great to use, but I have no idea how to install it or what I need to install to use it first.

**Reply by gwaldron** [reply]
If you are building osgEarth on your own, make sure you have the CMake flag `OSGEARTH_BUILD_TOOLS` toggled on. Then osgearth_conv will build into your install folder.

If you are installing osgEarth from vcpkg, you have to install the osgearth[tools] port to get it, e.g.
```
vcpkg install osgearth[tools]:x64-windows
```
[Look here for some basic information on osgearth_conv usage](https://docs.osgearth.org/en/latest/data.html#building-tilesets).

**Reply by nixongr14** [reply]
Sorry Glenn you beat me to it. I was going to delete my message above, because I was able to install. 

My last question really now is how do I convert .osm to mbtiles. I'm assuming I would just switch the argument to in-osm

**Comment by jasonbeverage** [comment]
What Glenn's command would do is actually download a small subset of the actual OSM raster tiles to an mbtiles dataset that you could then add to your earth file with:
```
<MBTilesImage>
    <url>osm.mbtiles</url>
</MBTilesImage>
```
This would allow you to access a small subset of the rendered OpenStreetMap service while you are offline.

Converting a raw .osm to mbtiles is something that is out of scope of what osgearth does.  .osm files contain actual vector geometry and would need to go through some kind of styling process (like what the OpenStreetMap online service is doing) before it's actually rendered to an image.

You can read about how you might set up your own OpenStreetMap tile server [here](https://switch2osm.org/serving-tiles/).
